ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -511,6 +516,39 @@ public void configureNextLayoutAnimation(JObject config, ICallback success, ICal
 
         #endregion
 
+        #region Sync React Methods
+
+        /// <summary>
+        /// Gets the constants for the view manager.
+        /// </summary>
+        /// <param name="viewManagerName">The view manager name.</param>
+        /// <returns>The view manager constants.</returns>
+        [ReactMethod(IsBlockingSynchronousMethod = true)]
+        public IDictionary<string, object> getConstantsForViewManager(string viewManagerName)
+        {
+            var viewManager = _uiImplementation.ResolveViewManager(viewManagerName);
+
```

Comment:
Threading concern: the backing dictionary in ViewManagerRegistry is a regular Dictionary. As long as creation of view manager is NOT lazy (and it seems they are not, we create all of them on startup) this call is safe. Perhaps we should add a comment somewhere...

Reply:
Agreed - lets add a comment (and convert the stored dictionary to IReadOnlyDictionary)

---
In reply to: [174852065](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174852065) [](ancestors = 174852065)